Assets/Scripts/Player/PlayerCharacter.cs

```diff
@@ -9,39 +10,37 @@ public class PlayerCharacter : MonoBehaviour
 {
     public WeaponSystem WeaponSystem { private get; set; }
     public HealthSystem HealthSystem { private get; set; }
+
+    public bool IsBoostedSpeed{ private get; set; }
+    
+    ///temporary? for working with skills
+    public Weapon PlayerWeapon { get; private set; }
+
     public GameState GameState { get; set; }
-    public int CountBullets{ get; set; }
+    public int CountBullets{ get; private set; }
 
-    public int PlayerClip => _playerWeapon.CountBulletInTheClip;
-    public int MaxHealthPlayer => HealthSystem.MaxHeals;
-    public int CurrentHealthPlayer => HealthSystem.Health;
+    private int MaxHealthPlayer => HealthSystem.MaxHeals;
+    private int CurrentHealthPlayer => HealthSystem.Health;
     
     [SerializeField] private LayerMask layerEnemy;
     [SerializeField] private Transform targetForLook;
     [SerializeField] private  List<GameObject> prefabsWeapon;
 
     private float _speed;
     private float _boostSpeedRate;
-    public GameObject _gunEquipped{ get; private set; }
-    private Weapon _playerWeapon;
-    private List<GameObject> _listGun;
-    private int CountGun => _listGun.Count;
-    private int _indexGun = 0;
-    // temporarily
-    private int _countBulletInTheClip;
     
     private void Update()
     {
         if (Input.GetKeyDown(KeyCode.Q))
         {
             //Move to Weapon system
-            _playerWeapon.ReturnAllBulletToSpawn();
-            _playerWeapon = WeaponSystem.SwitchWeapon();
+            PlayerWeapon.ReturnAllBulletToSpawn();
+            PlayerWeapon = WeaponSystem.SwitchWeapon();
```

Comment:
Pay close attention to your role separation, the player does not need to know that the bullets need to be returned to spawn, that is the responsibility of WeaponSystem, so it show look like this
```suggestion
            PlayerWeapon = WeaponSystem.SwitchWeapon();
```